source/multigrid/mg_tools.cc

```diff
@@ -1547,6 +1547,32 @@ namespace MGTools
       }
 
   }
+
+
+
+  template <int dim, int spacedim>
+  unsigned int
+  max_level_for_coarse_mesh (const Triangulation<dim,spacedim> &tria)
+  {
+    // Find minimum level for an active cell in
+    // this locally owned subdomain
+    int min_level = tria.n_global_levels();
+    typename Triangulation<dim,spacedim>::active_cell_iterator cell = tria.begin_active(),
+                                                               endc = tria.end();
+    for (; cell!=endc; ++cell)
+      if (cell->is_locally_owned())
+        if (cell->level() < min_level)
+          min_level = cell->level();
```

Comment:
You can actually `break` the outer loop here. This is because we traverse cells level by level, so if you have found a cell that is locally owned, then you don't need to search any further -- you're not going to find anything with an even lower level after this one.

Reply:
Ah, I didn't realize that. good point!

Reply:
Could we then skip the `<` check? If we traverse by level we could just say
`for (; cell!=endc; ++cell)`
    `    if (cell->is_locally_owned())`
    `    {`
       `        min_level = cell->level();`
        `        break;`
    `    } `

Reply:
Yes, you can do that. But you still need to initialize `min_level` as above to make sure you can deal with the case where a processor has no locally owned cells.